"Hibernate Reactive with PANACHE" No annotation @ProjectedFieldName

### Describe the bug

When I only use Hibernate Reactive, there is no @ProjectedFieldName in the imported jar, resulting in a compilation error; when I import the jar package "quarkus-hibernate-orm-panache-common" that contains @ProjectedFieldName, another error occurs:

```
Caused by: io.quarkus.runtime.configuration.ConfigurationException: Unable to find a JDBC driver corresponding to the database kind 'mysql' for the default datasource. Either provide a suitable JDBC driver extension, define the driver manually, or disable the JDBC datasource by adding 'quarkus.datasource.jdbc=false' to your configuration if you don't need it.
	at io.quarkus.agroal.deployment.AgroalProcessor.resolveDriver(AgroalProcessor.java:334)
	at io.quarkus.agroal.deployment.AgroalProcessor.getAggregatedConfigBuildItems(AgroalProcessor.java:280)
	at io.quarkus.agroal.deployment.AgroalProcessor.build(AgroalProcessor.java:85)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at io.quarkus.deployment.ExtensionLoader$2.execute(ExtensionLoader.java:820)
	at io.quarkus.builder.BuildContext.run(BuildContext.java:277)
	at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2449)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1478)
	at java.base/java.lang.Thread.run(Thread.java:834)
	at org.jboss.threads.JBossThread.run(JBossThread.java:501)
```

Is it necessary to put @ProjectedFieldName in "quarkus-hibernate-reactive-panache-common"？

### Expected behavior

_No response_

### Actual behavior

_No response_

### How to Reproduce?

_No response_

### Output of `uname -a` or `ver`

Microsoft Windows [Version 10.0.22000.168]

### Output of `java -version`

java version "11.0.10" 2021-01-19

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

2.2.2.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.8.1

### Additional information

_No response_

--- a/extensions/panache/hibernate-reactive-panache-common/runtime/src/main/java/io/quarkus/hibernate/reactive/panache/common/ProjectedFieldName.java
+++ b/extensions/panache/hibernate-reactive-panache-common/runtime/src/main/java/io/quarkus/hibernate/reactive/panache/common/ProjectedFieldName.java
@@ -1,0 +1,17 @@
+package io.quarkus.hibernate.reactive.panache.common;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Define a field's path for the SELECT statement when using a projection DTO.
+ * It supports the "dot" notation for fields in referenced entities:
+ * the name is composed of the property name for the relationship, followed by a dot ("."), followed by the name of the field.
+ */
+@Retention(RetentionPolicy.RUNTIME)
+@Target(ElementType.PARAMETER)
+public @interface ProjectedFieldName {
+    String value();
+}

--- a/extensions/panache/hibernate-reactive-panache-common/runtime/src/main/java/io/quarkus/hibernate/reactive/panache/common/runtime/CommonPanacheQueryImpl.java
+++ b/extensions/panache/hibernate-reactive-panache-common/runtime/src/main/java/io/quarkus/hibernate/reactive/panache/common/runtime/CommonPanacheQueryImpl.java
@@ -15,6 +15,7 @@ import org.hibernate.Filter;
 import org.hibernate.internal.util.LockModeConverter;
 import org.hibernate.reactive.mutiny.Mutiny;
 
+import io.quarkus.hibernate.reactive.panache.common.ProjectedFieldName;
 import io.quarkus.panache.common.Page;
 import io.quarkus.panache.common.Range;
 import io.quarkus.panache.common.exception.PanacheQueryException;
@@ -78,16 +79,24 @@ public class CommonPanacheQueryImpl<Entity> {
         StringBuilder select = new StringBuilder("SELECT new ").append(type.getName()).append(" (");
         int selectInitialLength = select.length();
         for (Parameter parameter : constructor.getParameters()) {
-            if (!parameter.isNamePresent()) {
+            String parameterName;
+            if (parameter.isAnnotationPresent(ProjectedFieldName.class)) {
+                final String name = parameter.getAnnotation(ProjectedFieldName.class).value();
+                if (name.isEmpty()) {
+                    throw new PanacheQueryException("The annotation ProjectedFieldName must have a non-empty value.");
+                }
+                parameterName = name;
+            } else if (!parameter.isNamePresent()) {
                 throw new PanacheQueryException(
                         "Your application must be built with parameter names, this should be the default if" +
                                 " using Quarkus artifacts. Check the maven or gradle compiler configuration to include '-parameters'.");
+            } else {
+                parameterName = parameter.getName();
             }
-
             if (select.length() > selectInitialLength) {
                 select.append(", ");
             }
-            select.append(parameter.getName());
+            select.append(parameterName);
         }
         select.append(") ");
 

--- a/integration-tests/hibernate-reactive-panache/src/main/java/io/quarkus/it/panache/reactive/Cat.java
+++ b/integration-tests/hibernate-reactive-panache/src/main/java/io/quarkus/it/panache/reactive/Cat.java
@@ -8,10 +8,16 @@ import io.quarkus.hibernate.reactive.panache.PanacheEntity;
 @Entity
 public class Cat extends PanacheEntity {
 
+    String name;
     @ManyToOne
     CatOwner owner;
 
     public Cat(CatOwner owner) {
+        this.owner = owner;
+    }
+
+    public Cat(String name, CatOwner owner) {
+        this.name = name;
         this.owner = owner;
     }
 

--- a/integration-tests/hibernate-reactive-panache/src/main/java/io/quarkus/it/panache/reactive/CatDto.java
+++ b/integration-tests/hibernate-reactive-panache/src/main/java/io/quarkus/it/panache/reactive/CatDto.java
@@ -1,0 +1,18 @@
+package io.quarkus.it.panache.reactive;
+
+import io.quarkus.hibernate.reactive.panache.common.ProjectedFieldName;
+import io.quarkus.runtime.annotations.RegisterForReflection;
+
+@RegisterForReflection
+public class CatDto {
+
+    public String name;
+
+    public String ownerName;
+
+    public CatDto(String name, @ProjectedFieldName("owner.name") String ownerName) {
+        this.name = name;
+        this.ownerName = ownerName;
+    }
+
+}

--- a/integration-tests/hibernate-reactive-panache/src/main/java/io/quarkus/it/panache/reactive/TestEndpoint.java
+++ b/integration-tests/hibernate-reactive-panache/src/main/java/io/quarkus/it/panache/reactive/TestEndpoint.java
@@ -1454,7 +1454,7 @@ public class TestEndpoint {
 
     @ReactiveTransactional
     @GET
-    @Path("projection")
+    @Path("projection1")
     public Uni<String> testProjection() {
         return Person.count()
                 .flatMap(count -> {
@@ -1493,6 +1493,25 @@ public class TestEndpoint {
 
                                 return "OK";
                             });
+                });
+    }
+
+    @ReactiveTransactional
+    @GET
+    @Path("projection2")
+    public Uni<String> testProjection2() {
+        String ownerName = "Julie";
+        String catName = "Bubulle";
+        CatOwner catOwner = new CatOwner(ownerName);
+        return catOwner.persist()
+                .chain(() -> new Cat(catName, catOwner).persist())
+                .chain(() -> Cat.find("name", catName)
+                        .project(CatDto.class)
+                        .<CatDto> firstResult())
+                .map(cat -> {
+                    Assertions.assertEquals(catName, cat.name);
+                    Assertions.assertEquals(ownerName, cat.ownerName);
+                    return "OK";
                 });
     }
 

--- a/integration-tests/hibernate-reactive-panache/src/test/java/io/quarkus/it/panache/reactive/PanacheFunctionalityTest.java
+++ b/integration-tests/hibernate-reactive-panache/src/test/java/io/quarkus/it/panache/reactive/PanacheFunctionalityTest.java
@@ -51,7 +51,8 @@ public class PanacheFunctionalityTest {
 
         RestAssured.when().get("/test/model1").then().body(is("OK"));
         RestAssured.when().get("/test/model2").then().body(is("OK"));
-        RestAssured.when().get("/test/projection").then().body(is("OK"));
+        RestAssured.when().get("/test/projection1").then().body(is("OK"));
+        RestAssured.when().get("/test/projection2").then().body(is("OK"));
         RestAssured.when().get("/test/model3").then().body(is("OK"));
     }
 

